tests/instrumentation/requests_tests.py

```diff
@@ -0,0 +1,57 @@
+import requests
+from requests.exceptions import InvalidURL, MissingSchema
+from urllib3.exceptions import ConnectionError
+
+import opbeat
+import opbeat.instrumentation.control
+from opbeat.traces import trace
+from tests.helpers import get_tempstoreclient
+from tests.utils.compat import TestCase
+
+
+class InstrumentRequestsTest(TestCase):
+    def setUp(self):
+        self.client = get_tempstoreclient()
+        opbeat.instrumentation.control.instrument()
+
+    def test_requests_instrumentation(self):
+        self.client.begin_transaction("transaction.test")
+        with trace("test_pipeline", "test"):
+            try:
+                requests.get('http://example.com')
```

Comment:
I understand that as a general rule, but what is the problem with this case? I don't care what is the result or which exception raises (i could catch a broader exception type...)

The thing with mocks is that i am not sure if/how mock instrumentation can interfere with Opbeat instrumentation and silently invalidate the result. I rather prefer to test the real thing...